module/VuFind/src/VuFind/Cover/Loader.php

```diff
@@ -557,44 +569,52 @@ protected function processImageURLForSource($url, $allowCache, $source)
      */
     protected function processImageURL($url, $cache = true)
     {
-        // Attempt to pull down the image:
-        $result = $this->client->setUri($url)->send();
-        if (!$result->isSuccess()) {
-            $this->debug("Failed to retrieve image from " + $url);
-            return false;
-        }
+        // Check to see if url is filepath MOD
+        if (substr($url, 0, 5) == "file:") {
+            $imagePath = substr($url, 5);
 
-        $image = $result->getBody();
+            // Display the image:
+            $this->contentType = 'image/jpeg';
```

Comment:
If memory serves, the API code below hard-codes this value to image/jpeg because the images get converted to a JPEG as part of the downloading process. If you're going to retrieve the image files from disk without modification here, you can't assume that they are JPEGs. You should either run a conversion routine or else detect the correct MIME type. I'd recommend testing this out with a GIF, for example, to see what happens. Let me know if you need me to look up the details of any of these operations.